js/app/modules/arrangement/sideBySide.js

```diff
@@ -51,65 +91,79 @@ const _SideBySideLayout = new Knowledge.Class({
         if (all_cards.length === 0)
             return [0, 0];
 
-        let [min, nat] = all_cards[0].get_preferred_width();
+        let [, first_nat] = all_cards[0].get_preferred_width();
+        let [, button_nat] = this._popover_button.get_preferred_width();
+
+        let min = this._left_padding * SPACING_MIN + first_nat + SPACING_MIN + button_nat + this._right_padding * SPACING_MIN;
+        let nat = first_nat;
 
         nat += all_cards.slice(1).reduce((accum, card) => {
             let [, card_nat] = card.get_preferred_width();
-            return accum + card_nat;
+            return accum + card_nat + SPACING_MAX;
         }, 0);
 
-        nat += _HorizontalSpacing.XLARGE * (all_cards.length - 1);
-        return [min, nat];
+        return [min, this._left_padding * SPACING_MAX + nat + this._right_padding * SPACING_MAX];
+    },
+
+    _ensure_card_parent: function (card, parent) {
+        let current_parent = card.get_parent();
+
+        if (current_parent !== parent) {
+            /* No need to add a reference sin GJS holds one for us */
+            current_parent.remove(card);
+            parent.add(card);
+            card.show();
+        }
     },
 
     vfunc_size_allocate: function (alloc) {
-        this.parent(alloc);
+        let all_cards = this.get_parent().get_cards();
 
         this.all_visible = true;
 
-        if (this.get_children().length === 0)
+        if (all_cards.length === 0)
             return;
 
-        let all_cards = this.get_parent().get_cards();
+        let [, button_nat] = this._popover_button.get_preferred_width();
+        /* Use cache value if button_nat is 0 (_popover_button is not visible) */
+        this._button_nat = button_nat || this._button_nat;
+
+        let width = alloc.width;
+        let cards_width = 0;
+        let spacing = this._button_nat + SPACING_MIN + this._left_padding * SPACING_MIN + this._right_padding * SPACING_MIN;
+        let i = 0;
 
-        let cards_width = all_cards.reduce((accum, card) => {
+        all_cards.forEach ((card) => {
             let [, card_nat] = card.get_preferred_width();
-            return accum + card_nat;
-        }, 0);
 
-        let leftover = (alloc.width - cards_width) / (all_cards.length - 1);
-        let spacing = this._get_horizontal_spacing(leftover);
-        let x = alloc.x;
-        let y = alloc.y;
+            cards_width += card_nat;
 
-        all_cards.forEach((card) => {
-            let [, card_nat] = card.get_preferred_width();
-            if (card_nat <= alloc.width) {
-                let offset = card_nat + spacing;
-                Arrangement.place_card(card, x, y, card_nat, MENU_HEIGHT);
-                alloc.width -= offset;
-                x += offset;
-            } else {
+            if (this.all_visible && width >= cards_width + spacing)
+                this._ensure_card_parent(card, this._box);
```

Comment:
I'm not entirely sure about this but I think this also queues an allocation from within size_allocate. Maybe keep all the cards parented inside the arrangement but just `set_child_visible()`, and keep a separate set of all the cards parented inside the popover?